airbyte-cdk/bulk/core/load/src/test/kotlin/io/airbyte/cdk/message/MockStreamsManagerFactory.kt

```diff
@@ -67,8 +68,9 @@ class MockStreamManager : StreamManager {
     }
 }
 
-@Prototype
+@Singleton
 @Requires(env = ["MockStreamsManager"])
+@Replaces(StreamsManager::class)
```

Comment:
I've never been sure about `@Replaces`. Is it needed if the thing you're replacing is obviously the thing you're implementing? Does it resolve ambiguity ever?

Reply:
I still don't really grok why this was necessary - probably related to @jdpgrailsdev 's comment https://github.com/airbytehq/airbyte/pull/45884#discussion_r1775130257. Without this, micronaut was injecting the actual stream manager. (plausibly this also should just be `@Primary`? I don't recall why I went with `@Replaces` here, probably just random flailing)

Reply:
oh, after reading the docs jonathan linked - there's probably some redundant annotations. I think either:
* runtime code is all tagged `secondary` (+ maybe test code is tagged primary?)
* or runtime code has no annotation, and test code uses `replaces`

but (I think?) doing both secondary+replaces is unnecessary